jina/orchestrate/flow/base.py

```diff
@@ -2368,24 +2366,97 @@ def block(
+            watch_files_from_deployments = {}
+            for name, deployment in self._deployment_nodes.items():
+                if deployment.args.restart:
+                    if deployment._is_executor_from_yaml:
+                        watch_files_from_deployments[deployment.args.uses] = name
+            watch_files_list = list(watch_files_from_deployments.keys())
+
+            config_loaded = getattr(self, '_config_loaded', '')
+            if config_loaded.endswith('yml') or config_loaded.endswith('yaml'):
+                watch_files_list.append(config_loaded)
+
+            if watch_changes and len(watch_files_list) > 0:
+
+                with ImportExtensions(
+                    required=True,
+                    logger=self.logger,
+                    help_text='''restart requires watchfiles dependency to be installed. You can do `pip install 
+                    watchfiles''',
+                ):
+                    from watchfiles import watch
+
+                new_stop_event = stop_event or threading.Event()
+                if len(watch_files_list) > 0:
                     while True:
-                        if self._stop_event.is_set():
+                        for changes in watch(
+                            *watch_files_list, stop_event=new_stop_event
+                        ):
+                            for _, changed_file in changes:
+                                if changed_file not in watch_files_from_deployments:
+                                    # maybe changed_file is the absolute path of one in watch_files_from_deployments
+                                    is_absolute_path = False
+                                    for (
+                                        file,
+                                        deployment_name,
+                                    ) in watch_files_from_deployments.items():
+                                        if changed_file.endswith(file):
+                                            is_absolute_path = True
+                                            _restart_deployment(
+                                                self._deployment_nodes[deployment_name],
+                                                changed_file,
+                                            )
+                                            break
+
+                                    if not is_absolute_path:
+                                        _restart_flow(changed_file)
+                                else:
+                                    _restart_deployment(
+                                        self._deployment_nodes[
+                                            watch_files_from_deployments[changed_file]
+                                        ],
+                                        changed_file,
+                                    )
+                        if new_stop_event.is_set():
                             break
-                        time.sleep(0.5)
```

Comment:
to be removed as well

Reply:
no, then ur CPU will be burning, we need it to rest